ENIGMAsystem/SHELL/Platforms/General/POSIX/BSD/BSDprogdir.cpp

```diff
@@ -0,0 +1,50 @@
+#include "Platforms/General/POSIX/POSIXprogdir.h"
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#include <cstddef>
+#include <string>
+
+using std::string;
+using std::size_t;
+
+static inline string program_directory_helper(size_t length) {
+  string path;
+  length++;
+  int mib[4];
+  mib[0] = CTL_KERN;
+  mib[1] = KERN_PROC;
+  mib[2] = KERN_PROC_PATHNAME;
+  mib[3] = -1;
+  char *buffer = path.data();
+  size_t cb = length;
+  if (sysctl(mib, 4, buffer, &cb, NULL, 0) == -1) {
+    path = program_directory_helper(length);
+  } else if (sysctl(mib, 4, buffer, &cb, NULL, 0) == 0) {
+    path = enigma_user::filename_path(buffer);
+  }
+  return path;
+}
+
+namespace enigma {
+
+  void initialize_program_directory() {
+    string path;
+    size_t length = 256;
+    int mib[4];
+    mib[0] = CTL_KERN;
+    mib[1] = KERN_PROC;
+    mib[2] = KERN_PROC_PATHNAME;
+    mib[3] = -1;
+    char *buffer = new char[length]();
```

Comment:
as previously mentioned you can just assign to underlying data of the std::string instead of allocating your own buffer here

Reply:
I'm doing it correctly, so there's no reason to listen to you on this one. Marking it resolved. Have a good day. <3

Reply:
don't resolve conversations unless both parties agree on solution

Reply:
I'll do it as long as you state things I'm already aware of because you've said it 1000x on discord and only say it on here to carry over that same drama to a different website.